tools/dnt_preset_editor.html

```diff
@@ -1139,6 +1338,695 @@
+            if (param.allControlOptions && param.allControlOptions.length > 0) {
+                drawCalls.push(`    drawText(${cX}, ${rY}, string.format("%s: %s", "${param.nickname.toUpperCase()}", getEnumText(${nickname}_options, ${nickname}_value)))`);
+                displayLine = `${param.nickname.toUpperCase()}: [Enum]`;
+            } else {
+                if (param.unit === 'dB') {
+                    drawCalls.push(`    local ${nickname}_db = to_db(${nickname}_value or 0)`);
+                    drawCalls.push(`    drawText(${cX}, ${rY}, string.format("%s: %3.1fdB", "${param.nickname.toUpperCase()}", ${nickname}_db))`);
+                    displayLine = `${param.nickname.toUpperCase()}: dB value`;
+                } else {
+                    drawCalls.push(`    local ${nickname}_percent = (${nickname}_value or 0) * 100`);
+                    drawCalls.push(`    drawText(${cX}, ${rY}, string.format("%s: %3.1f", "${param.nickname.toUpperCase()}", ${nickname}_percent))`);
+                    displayLine = `${param.nickname.toUpperCase()}: % value`;
+                }
+            }
+
+            // add to drawCommands for preview
+            let cXVal = [10, 74, 138][colIndex];
+            let rYVal;
+            switch (rowIndex) {
+                case 0: rYVal = 20; break;
+                case 1: rYVal = 35; break;
+                case 2: rYVal = 50; break;
+                case 3: rYVal = 65; break;
+            }
+
+            drawCommands.push({
+                type: 'text',
+                x: cXVal,
+                y: rYVal,
+                text: displayLine,
+                shade: 15,
+                paramIndex: paramIndexForHighlight
+            });
+        }
+
+        if (globalParamList.length > 12) {
+            drawCalls.push(`    -- Note: More than 12 parameters are mapped but only 12 are displayed.`);
+            drawCommands.push({
+                type: 'text',
+                x: 10,
+                y: 60,
+                text: "Only 12 displayed",
+                shade: 8
+            });
+        }
+
+        // Build final Lua script
+        let finalScript = [];
+        finalScript.push('return {');
+        finalScript.push(`    name = '${presetName} UI script',`);
+        finalScript.push(`    author = 'chordsmaze',`);
+        finalScript.push(`    description = 'UI script for ${presetName} preset created by the UI Builder by chordsmaze',`);
+        finalScript.push('');
+        finalScript.push('    init = function()');
+        finalScript.push('        -- Get initial parameter values');
+        initCode.forEach(line => finalScript.push(line));
+        finalScript.push('        return true');
+        finalScript.push('    end,');
+        finalScript.push('');
+        controllerFuncs.forEach(line => finalScript.push(line));
+        if (!controllerFuncs.some(c => c.includes('button4Push'))) {
+            finalScript.push('    button4Push = function()');
```

Comment:
I would prefer this was button 2, consistent with the other UI scripts in this repo.

Reply:
Sounds good, I noticed I’ve been hitting button 4 accidently when turning pot 3